### ISSUE TITLE
Problems of padding and cropping

### ISSUE THREAD

**Issue by WMCh** [baowenbo/DAIN#51]
Hi.Thanks for the wonderful work!

I ran the test code on my `1280*720` video but I noticed that the edges of the image were moving, and it looked like a few pixels had been vertically compressed (720 is not divisible by 128). I see that your code has a padding operation for the input image and a clipping operation for the output image, but I don't know what caused this boundary drift problem.

Can you help me? Thanks!

**Comment by WMCh** [comment]
I added some tests, including removing the padding operation and resizing the image size, which basically eliminated the assumption that the compression problem of generated frames was caused by padding. And I found the same offset problem in the provided test set (MiddleBurySet). So I now further suspect that the model itself caused the drift problem. What do you think?

**Comment by jmspiewak** [comment]
Hi, I think it is caused by an anomaly in the pretrained PWCNet model. For some reason even when you pass two identical images it reports everything drifting inwards.

A quick workaround I found is adjusting the flow from A to B by the drift between A and itself.
In forward_flownets:
```python
im1 = input[:,:3,:,:]
drift = model(torch.cat([im1, im1], dim=1))
temp = model(input) - drift
```

The drift is still there, but much less noticeable. At least from the quick testing I did.

**Comment by htleeab** [comment]
The workaround of @jmspiewak works for me.
If the drift issue is caused by the pretrained PWCNet model, do it mean re-training model could fix this issue? or is PWCNet resolution dependent?

**Comment by mpriessner** [comment]
I had the same problem in my implementation on Colab where I use miniconda for DAIN 
https://towardsdatascience.com/conda-google-colab-75f7c867a522
and I use the implementation of iBobbyTS:https://github.com/iBobbyTS/VFIN 

When I changed from pytorch 1.4 to pytorch 1.1 it started working without the padding problem (with the GPU P100). 
!conda install pytorch==1.4.0 torchvision==0.5.0 cudatoolkit=10.1 -c pytorch -y 
to 
!conda install pytorch==1.1.0 torchvision==0.3.0 cudatoolkit=10.0 -c pytorch  -y
I hope that helps some of you.

**Comment by iBobbyTS** [comment]
> I had the same problem in my implementation on Colab where I use miniconda for DAIN 
> 
> https://towardsdatascience.com/conda-google-colab-75f7c867a522
> 
> and I use the implementation of iBobbyTS:https://github.com/iBobbyTS/VFIN 
> 
> 
> 
> When I changed from pytorch 1.4 to pytorch 1.1 it started working without the padding problem (with the GPU P100). 
> 
> !conda install pytorch==1.4.0 torchvision==0.5.0 cudatoolkit=10.1 -c pytorch -y 
> 
> to 
> 
> !conda install pytorch==1.1.0 torchvision==0.3.0 cudatoolkit=10.0 -c pytorch  -y
> 
> I hope that helps some of you.
> 
> 

Can you post the exact error message? 

By the way, usually conda doesn't work in colab, unless you do %%bash so that you will have a proper shell environment. So I would recommend using original Python instead of anaconda or miniconda. 
For VFIN, did you try my built Python and DAIN package? I also have a cell with all the codes that generates the built package.

**Comment by linyu0219** [comment]
> Hi, I think it is caused by an anomaly in the pretrained PWCNet model. For some reason even when you pass two identical images it reports everything drifting inwards.
> 
> A quick workaround I found is adjusting the flow from A to B by the drift between A and itself.
> In forward_flownets:
> 
> ```python
> im1 = input[:,:3,:,:]
> drift = model(torch.cat([im1, im1], dim=1))
> temp = model(input) - drift
> ```
> 
> The drift is still there, but much less noticeable. At least from the quick testing I did.

It worked !
Maybe @styler00dollar   @AlphaGit  can update colab.
By the way colab use pytorch 1.4 will have erros,I use my local env at last.
my source video is 256*256 30fps->60fps, then I find the shift.
@iBobbyTS I don't try your DAIN,but i find ssm has same problems.
